nixos/jormungandr.nix

```diff
@@ -191,9 +202,12 @@ in {
           };
           p2p = {
             public_address = cfg.publicAddress;
+
             trusted_peers = cfg.trustedPeersAddresses;
             topics_of_interest = cfg.topicsOfInterest;
-          } // (if (cfg.publicId != null) then {
+          } // (if (cfg.listenAddress != null) then {
```

Comment:
I think we could use `lib.optionalAttrs` here, in general.

Reply:
yeah I'm going to try again. I remember having issues where `toJson` was serializing the thunk of the function instead of fully evaluating it.

Reply:
Ah, I see, there's some complicating context -- maybe it's not worth the trouble then, @jbgi, your call, really.